### PR TITLE
[Merged by Bors] - proposals: validate that referenced atx is included into active set

### PR DESCRIPTION
**Pull request by dshulyak**
related: https://github.com/spacemeshos/go-spacemesh/issues/3936

### CODE REVIEW

File: proposals/eligibility_validator.go [hunk 1]
```diff
@@ -80,19 +83,29 @@ func (v *Validator) CheckEligibility(ctx context.Context, ballot *types.Ballot)
...
-	if err != nil {
-		return false, fmt.Errorf("get ballot ATX header %v: %w", ballot.AtxID, err)
+	if owned == nil {
+		return false, fmt.Errorf("atx %v from ballot %v (refballot %v) is not included into the active set", ballot.AtxID, ballot.ID(), refBallot.ID())
+	}
+	if targetEpoch := owned.TargetEpoch(); targetEpoch != epoch {
+		return false, fmt.Errorf("%w: ATX target epoch (%v), ballot publication epoch (%v)",
+			errTargetEpochMismatch, targetEpoch, epoch)
+	}
+	if pub := ballot.SmesherID(); !bytes.Equal(owned.NodeID.Bytes(), pub.Bytes()) {
```
Comment (countvonzero) [inline proposals/eligibility_validator.go:107]: you can do "==" with types.NodeID (types.Hash32)

**Review by countvonzero — APPROVED** [review]

File: proposals/eligibility_validator.go (hunk 1 above)
Comment (dshulyak) [reply proposals/eligibility_validator.go:107]: copied this

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by countvonzero** [comment]
bors merge

**Comment by dshulyak** [comment]
@countvonzero just fyi, bors applies changes on top before merge, if there are no conflicts rebasing and stuff is not necessary

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by countvonzero** [comment]
> @countvonzero just fyi, bors applies changes on top before merge, if there are no conflicts rebasing and stuff is not necessary

this PR was not in the bors queue when i merge the develop branch. you are saying if i do "bors merge" when its not in the queue, it will still resolve conflicts? good to know.

btw, the failing unit test seems legit from the new code (after i merged develop).

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**